google-cloud-firestore/lib/google/cloud/firestore/query.rb

```diff
@@ -0,0 +1,613 @@
+        #
+        #   query.get do |city|
+        #     puts "#{city.document_id} has #{city[:population]} residents."
+        #   end
+        #
+        def order name, direction = :asc
+          new_query = @query.dup
+          new_query ||= StructuredQuery.new
+
+          new_query.order_by << StructuredQuery::Order.new(
+            field: StructuredQuery::FieldReference.new(field_path: name.to_s),
+            direction: order_direction(direction))
+
+          Query.start new_query, parent_path, client
+        end
+        alias_method :order_by, :order
+
+        ##
+        # Skips to an offset in a query. If the current query already has
+        # specified an offset, this will overwrite it.
+        #
+        # @param [Integer] num The number of results to skip.
+        #
+        # @return [Query] New query with `offset` called on it.
+        #
+        # @example
+        #   require "google/cloud/firestore"
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #
+        #   # Get a collection reference
+        #   cities_col = firestore.col "cities"
+        #
+        #   # Create a query
+        #   query = cities_col.limit(5).offset(10)
+        #
+        #   query.get do |city|
+        #     puts "#{city.document_id} has #{city[:population]} residents."
+        #   end
+        #
+        def offset num
+          new_query = @query.dup
+          new_query ||= StructuredQuery.new
+
+          new_query.offset = num
+
+          Query.start new_query, parent_path, client
+        end
+
+        ##
+        # Limits a query to return a fixed number of results. If the current
+        # query already has a limit set, this will overwrite it.
+        #
+        # @param [Integer] num The maximum number of results to return.
+        #
+        # @return [Query] New query with `limit` called on it.
+        #
+        # @example
+        #   require "google/cloud/firestore"
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #
+        #   # Get a collection reference
+        #   cities_col = firestore.col "cities"
+        #
+        #   # Create a query
+        #   query = cities_col.offset(10).limit(5)
+        #
+        #   query.get do |city|
+        #     puts "#{city.document_id} has #{city[:population]} residents."
+        #   end
+        #
+        def limit num
+          new_query = @query.dup
+          new_query ||= StructuredQuery.new
+
+          new_query.limit = Google::Protobuf::Int32Value.new(value: num)
+
+          Query.start new_query, parent_path, client
+        end
+
+        ##
+        # Starts query results at a particular document value. The result set
+        # will include the document specified by `values`.
+        #
+        # If the current query already has specified `start_at` or
+        # `start_after`, this will overwrite it.
+        #
+        # When the query is sent to the server, the `values` will be used in
```

Comment:
Can you reword this to more clearly state the relationship? I am not sure that I would understand it from this comment.

Reply:
As I understand it, when you set a cursor using field values, the values are matched to the field names the query is set it to order by. Is that correct?

@quartzmo, can you help us in working this better?

Reply:
This has been updated. It now says:

> The values provided here are for the field paths provides to `order`. Values provided to `start_after` without an associated field path provided to `order` will result in an error.

What do you think?